src/ct_selftest/ct_selftest.cpp

```diff
@@ -209,6 +209,61 @@ void test_scoped_poison_outer(Botan::RandomNumberGenerator& rng) {
    }
 }
 
+/**
+ * @brief Test that the return value of poison/unpoison is correctly handled.
+ *
+ * In theory we only test unpoison here. Since the logic of poison and unpoison is
+ * equal we expect poison to work as well.
+ */
+void test_unpoison_inout(Botan::RandomNumberGenerator& rng) {
+   const uint8_t poisoned_byte = rng.next_byte();
+   Botan::CT::poison(poisoned_byte);
+
+   // Test that the return value is unpoisoned
+   if(Botan::CT::unpoison(poisoned_byte) == 0) {
+      std::cout << "I may or may not be printed." << std::endl;
+   }
+
+   // Test that the value is unpoisoned
+   if(poisoned_byte == 1) {
+      std::cout << "I may or may not be printed too." << std::endl;
+   }
+
+   auto poisoned_range = rng.random_vec(8);
+   Botan::CT::poison_range(poisoned_range);
```

Comment:
No need to test the range overload here. Just move this to `test_ct_utils.cpp`. Same rationale as explained for the `const_time_poison()` overloads.

Reply:
Alright :)